src/piece_picker.cpp

```diff
@@ -333,6 +333,19 @@ namespace libtorrent::aux {
 		*zero_prio = int(m_downloads[piece_pos::piece_zero_prio].size());
 	}
 
+    void piece_picker::set_sequential_range(piece_index_t const start_piece, piece_index_t const end_piece)
```

Comment:
Both Oleg, and Denis (in #4408), indicated the motivation for this API change for them as far as their uses cases are concerned is to be able to seek to a position in the media where they can start playing, or where the metadata is located so it can be parsed correctly. When you suggested an end piece they either accepted it as a nice to have or didn't care for it.

Reply:
and I think if they want to download sequentially until the last piece, they can specify that. but I don't think it's a reasonable default.

If anything, I would think a more reasonable default would be the end of the file that `start_piece` happens to fall in. That can get a bit tricky with pieces spanning multiple files though.

Reply:
I reckon it best to keep things at piece level addressability.